src/main/java/world/bentobox/islandfly/listeners/FlyListener.java

```diff
@@ -39,17 +40,14 @@ public FlyListener(final IslandFlyAddon addon) {
      */
     @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
     public void onExitIsland(final IslandEvent.IslandExitEvent event) {
-        // Check only when player exit is own island
-        // Player already flying
+
         final User user = User.getInstance(event.getPlayerUUID());
 
-        if (!addon.getIslands().userIsOnIsland(Util.getWorld(event.getLocation().getWorld()), user)) {
-            return;
-        }
-        
-        if (!user.getPlayer().getAllowFlight()) {
-            return;
-        }
+        final Optional<Island> i = addon.getIslands().getIslandAt(user.getLocation());
+
+        if (!i.isPresent()) return;
```

Comment:
Why not use `userIsOnIsland` like the original?

Reply:
My research told me that userIsOnIsland tells whether users is on his **own** island, but I added a feature that the /is fly is not limited only to his own island, but now can be used on other as well (depending whether the settings flag allows them)

Does that makes sense, or I just terribly explained?

Reply:
Ahhh, I see. Yes, in that case, it makes sense.